packages/editor/src/components/post-trash/check.js

```diff
@@ -18,29 +21,38 @@ import { store as editorStore } from '../../store';
  * @return {Component|null} The rendered child components or null if the post can not trashed.
  */
 export default function PostTrashCheck( { children } ) {
-	const { canTrashPost } = useSelect( ( select ) => {
-		const { isEditedPostNew, getCurrentPostId, getCurrentPostType } =
-			select( editorStore );
-		const { canUser } = select( coreStore );
-		const postType = getCurrentPostType();
-		const postId = getCurrentPostId();
-		const isNew = isEditedPostNew();
-		const canUserDelete = !! postId
-			? canUser( 'delete', {
-					kind: 'postType',
-					name: postType,
-					id: postId,
-			  } )
-			: false;
-
+	const { item, permissions, postType } = useSelect( ( select ) => {
+		const store = select( editorStore );
+		const { getEditedEntityRecord, getEntityRecordPermissions } = unlock(
+			select( coreStore )
+		);
+		const postId = store.getCurrentPostId();
+		const _postType = store.getCurrentPostType();
 		return {
-			canTrashPost: ( ! isNew || postId ) && canUserDelete,
+			item: getEditedEntityRecord( 'postType', _postType, postId ),
+			permissions: getEntityRecordPermissions(
+				'postType',
+				_postType,
+				postId
+			),
+			postType: _postType,
 		};
 	}, [] );
-
-	if ( ! canTrashPost ) {
+	const itemWithPermissions = useMemo( () => {
+		return {
+			...item,
+			permissions,
+		};
+	}, [ item, permissions ] );
+	const actions = usePostActions( { postType } );
```

Comment:
It feels way too heavy for me to call usePostActions just to check if one single action is illigible or not. I'd rather just duplicate the check.

Reply:
Hi @youknowriad, I updated the code and now we don't call usePostActions on eligibility check.